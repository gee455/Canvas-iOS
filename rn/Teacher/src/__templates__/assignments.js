//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const assignment: Template<Assignment> = template({
  id: '123456789',
  course_id: '987654321',
  name: 'learn to write code',
  description: 'a course that can take any beginner to professional in a matter of microseconds.  Harness the power of your favorite languages like PHP and Cold Fusion.',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
  created_at: '2017-03-17T19:15:25Z',
  updated_at: '2017-03-17T19:15:25Z',
  points_possible: 20,
  grading_type: 'points',
  assignment_group_id: 1376,
  peer_reviews: false,
  automatic_peer_reviews: false,
  position: 1,
  submission_types:
  [ 'online_text_entry',
    'online_url',
    'media_recording',
    'online_upload' ],
  needs_grading_count: 0,
  html_url: 'http://hostname.tld/whatever',
  overrides: [],
  grade_group_students_individually: false,
  group_category_id: null,
  has_overrides: false,
  only_visible_to_overrides: false,
  published: true,
  unpublishable: false,
  rubric_assessment: {},
  rubric_settings: null,
  rubric: null,
})

export const assignmentGroup: Template<AssignmentGroup> = template({
  id: '1',
  name: 'Learn React Native',
  position: 1,
  assignments: [assignment()],
  group_weight: 1,
  integration_data: {},
  sis_source_id: '',
})

export const assignmentDueDate: Template<AssignmentDate> = template({
  title: 'due',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
})

export const assignmentOverride: Template<AssignmentOverride> = template({
  id: '1',
  assignment_id: '1',
  student_ids: [],
  group_id: '1',
  course_section_id: '1',
  title: 'override',
  all_day: false,
  all_date_date: '',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
})
