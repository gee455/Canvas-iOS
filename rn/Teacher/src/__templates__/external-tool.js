//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const externalTool: Template<AddressBookResult> = template({
  id: '1',
  name: 'LTI FTW',
  url: 'https://canvas.instructure.com/lti',
})

export const ltiLaunchDefinition: Template<LtiLaunchDefinition> = template({
  definition_id: 42,
  placements: {
    course_navigation: {
      url: 'https://rollcall.instructure.com/launchy-launch',
    },
  },
})
