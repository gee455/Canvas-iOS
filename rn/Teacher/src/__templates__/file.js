//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const usageRights: Template<UsageRights> = template({
  legal_copyright: '',
  use_justification: 'own_copyright',
})

export const file: Template<File> = template({
  id: '111',
  display_name: 'Book Report',
  url: 'http://canvaslms.com/bookreport',
  size: 488212,
  thumbnail_url: 'http://fillmurray.com/322/200',
  mime_class: 'document',
  preview_url: '/',
  filename: '/bookreport',
  parent_folder_id: null,
  locked: false,
  hidden: false,
  unlock_at: null,
  lock_at: null,
  usage_rights: usageRights(),
  'content-type': 'application/document',
})
