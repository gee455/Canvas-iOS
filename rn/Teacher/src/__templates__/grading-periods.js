//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const gradingPeriod: Template<GradingPeriod> = template({
  id: '1023',
  title: 'First Block',
  start_date: '2014-01-07T15:04:00Z',
  end_date: '2014-05-07T17:07:00Z',
  close_date: '2014-06-07T17:07:00Z',
  weight: 33.33,
})
