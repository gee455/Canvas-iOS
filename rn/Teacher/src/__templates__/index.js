//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

export * from '../redux/__templates__/app-state'
export * from './account-notification'
export * from './addressBook'
export * from './api'
export * from './assignments'
export * from './attachment'
export * from './conversations'
export * from './course'
export * from './discussion'
export * from './enrollments'
export * from './error'
export * from './external-tool'
export * from './file'
export * from './folder'
export * from './grading-periods'
export * from './group'
export * from './helm'
export * from './launch-definitions'
export * from './mediaComment'
export * from './page'
export * from './quiz'
export * from './quizSubmission'
export * from './role'
export * from './rubric'
export * from './section'
export * from './session'
export * from './submissions'
export * from './tab'
export * from './toDo'
export * from './users'
