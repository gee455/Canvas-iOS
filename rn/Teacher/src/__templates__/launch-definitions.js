//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

import template, { type Template } from '../utils/template'

export const launchDefinitionGlobalNavigationItem: Template<ExternalToolLaunchDefinitionGlobalNavigationItem> = template({
  message_type: 'basic-lti-launch-request',
  url: 'https://mobiledev.beta.instructuremedia.com/lti/launch?custom_gauge_launch_type=global_nav',
  title: 'Gauge',
})

export const launchDefinition: Template<ExternalToolLaunchDefinition> = template({
  definition_type: 'ContextExternalTool',
  definition_id: '360860',
  name: 'Gauge',
  description: 'Tools for Education',
  domain: 'gauge.instructure.com',
  placements: {
    global_navigation: launchDefinitionGlobalNavigationItem(),
  },
})
