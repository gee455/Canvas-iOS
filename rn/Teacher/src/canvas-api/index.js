//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

import 'babel-polyfill'

export { default } from './apis/index'
export * from './apis/assignmentGroups'
export * from './apis/assignments'
export * from './apis/conversations'
export * from './apis/courses'
export * from './apis/discussions'
export * from './apis/enrollments'
export * from './apis/external-tools'
export * from './apis/groups'
export * from './apis/login'
export * from './apis/quizzes'
export * from './apis/submissions'
export * from './apis/users'
export * from './apis/media-comments'
export * from './apis/file-uploads'
export * from './apis/files'
export * from './apis/accounts'

export * from './session'

export { default as httpClient, isAbort, httpCache } from './httpClient'
