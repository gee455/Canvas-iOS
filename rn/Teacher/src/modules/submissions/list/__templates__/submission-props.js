//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow

import template, { type Template } from '../../../../utils/template'
import type { SubmissionDataProps } from '../submission-prop-types'

const templates = {
  ...require('../../../../__templates__/submissions'),
}

const submissionWithHistory = templates.submissionHistory()

// $FlowFixMe
export const submissionProps: Template<SubmissionDataProps> = template({
  grade: 'B-',
  name: submissionWithHistory.user.name,
  status: 'submitted',
  userID: submissionWithHistory.user.id,
  avatarURL: submissionWithHistory.user.avatar_url,
  sectionID: '1',
})
