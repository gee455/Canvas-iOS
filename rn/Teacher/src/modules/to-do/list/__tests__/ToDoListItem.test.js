//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

// @flow

import { shallow } from 'enzyme'
import React from 'react'
import { API, httpCache } from '../../../../canvas-api/model-api'
import * as template from '../../../../__templates__'
import Connected, { ToDoListItem } from '../ToDoListItem'

describe('ToDoListItem', () => {
  let props
  beforeEach(() => {
    httpCache.clear()
    props = {
      navigator: template.navigator(),
      item: template.toDoModel({ courseID: '1' }),
      courseName: 'Course 1',
      courseColor: '#fff',
      api: new API({ policy: 'cache-only' }),
      isLoading: false,
      loadError: null,
      refresh: jest.fn(),
      onPress: jest.fn(),
    }
  })

  it('gets courseColor and courseName from the model api', () => {
    const courseColor = 'green'
    const course = template.courseModel()
    httpCache.handle('GET', 'users/self/colors', { custom_colors: { course_1: courseColor } })
    httpCache.handle('GET', 'courses/1', course)
    const tree = shallow(<Connected item={props.item} />)
    expect(tree.find(ToDoListItem).props()).toMatchObject({
      courseColor,
      courseName: course.name,
    })
  })

  it('renders', () => {
    const tree = shallow(<ToDoListItem {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders published assignment', () => {
    props.item = template.toDoModel({
      assignment: template.assignment({ name: 'Assignment 1', published: true }),
    })
    const tree = shallow(<ToDoListItem {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders unpublished assignment', () => {
    props.item = template.toDoModel({
      assignment: template.assignment({ name: 'Assignment 1', published: false }),
    })
    const tree = shallow(<ToDoListItem {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders quiz', () => {
    props.item = template.toDoModel({
      assignment: null,
      quiz: template.quiz({ title: 'Quiz 1' }),
    })
    const tree = shallow(<ToDoListItem {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders discussion', () => {
    props.item = template.toDoModel({
      assignment: template.assignment({ name: 'Assignment 1', submission_types: ['discussion_topic'] }),
    })
    const tree = shallow(<ToDoListItem {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders null due date', () => {
    props.item = template.toDoModel({
      assignment: template.assignment({ due_at: null }),
    })
    const tree = shallow(<ToDoListItem {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('passes the item to onPress', () => {
    const tree = shallow(<ToDoListItem {...props} />)
    tree.find('Row').simulate('Press')
    expect(props.onPress).toHaveBeenCalledWith(props.item)
  })
})
