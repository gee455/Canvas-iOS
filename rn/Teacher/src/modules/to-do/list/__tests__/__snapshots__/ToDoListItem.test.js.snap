// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoListItem renders 1`] = `
<View>
  <Row
    accessible={true}
    disclosureIndicator={true}
    onPress={[Function]}
    renderImage={[Function]}
    testID="to-do.list.https://api.instructure.com/api/v1/path/to/html_url.row"
    title="learn to write code"
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 14,
              "fontWeight": "600",
            },
            Object {
              "color": "#fff",
            },
          ]
        }
      >
        Course 1
      </Text>
      <Text
        style={
          Object {
            "color": "#8B969E",
            "fontSize": 13,
            "marginTop": 3,
          }
        }
      >
        Due May 31, 2037 at 11:59 PM
      </Text>
      <Text
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "white",
              "borderRadius": 9,
              "borderWidth": 1,
              "flex": 0,
              "fontSize": 11,
              "fontWeight": "600",
              "marginTop": 4,
              "overflow": "hidden",
              "paddingBottom": 1,
              "paddingLeft": 6,
              "paddingRight": 6,
              "paddingTop": 3,
            },
            Object {
              "borderColor": "#374A59",
              "color": "#374A59",
            },
          ]
        }
      >
        5 NEED GRADING
      </Text>
    </View>
  </Row>
  <AccessLine
    disableAppSpecificChecks={false}
    visible={true}
  />
</View>
`;

exports[`ToDoListItem renders discussion 1`] = `
<View>
  <Row
    accessible={true}
    disclosureIndicator={true}
    onPress={[Function]}
    renderImage={[Function]}
    testID="to-do.list.https://api.instructure.com/api/v1/path/to/html_url.row"
    title="Assignment 1"
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 14,
              "fontWeight": "600",
            },
            Object {
              "color": "#fff",
            },
          ]
        }
      >
        Course 1
      </Text>
      <Text
        style={
          Object {
            "color": "#8B969E",
            "fontSize": 13,
            "marginTop": 3,
          }
        }
      >
        Due May 31, 2037 at 11:59 PM
      </Text>
      <Text
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "white",
              "borderRadius": 9,
              "borderWidth": 1,
              "flex": 0,
              "fontSize": 11,
              "fontWeight": "600",
              "marginTop": 4,
              "overflow": "hidden",
              "paddingBottom": 1,
              "paddingLeft": 6,
              "paddingRight": 6,
              "paddingTop": 3,
            },
            Object {
              "borderColor": "#374A59",
              "color": "#374A59",
            },
          ]
        }
      >
        5 NEED GRADING
      </Text>
    </View>
  </Row>
  <AccessLine
    disableAppSpecificChecks={false}
    visible={true}
  />
</View>
`;

exports[`ToDoListItem renders null due date 1`] = `
<View>
  <Row
    accessible={true}
    disclosureIndicator={true}
    onPress={[Function]}
    renderImage={[Function]}
    testID="to-do.list.https://api.instructure.com/api/v1/path/to/html_url.row"
    title="learn to write code"
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 14,
              "fontWeight": "600",
            },
            Object {
              "color": "#fff",
            },
          ]
        }
      >
        Course 1
      </Text>
      <Text
        style={
          Object {
            "color": "#8B969E",
            "fontSize": 13,
            "marginTop": 3,
          }
        }
      >
        No Due Date
      </Text>
      <Text
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "white",
              "borderRadius": 9,
              "borderWidth": 1,
              "flex": 0,
              "fontSize": 11,
              "fontWeight": "600",
              "marginTop": 4,
              "overflow": "hidden",
              "paddingBottom": 1,
              "paddingLeft": 6,
              "paddingRight": 6,
              "paddingTop": 3,
            },
            Object {
              "borderColor": "#374A59",
              "color": "#374A59",
            },
          ]
        }
      >
        5 NEED GRADING
      </Text>
    </View>
  </Row>
  <AccessLine
    disableAppSpecificChecks={false}
    visible={true}
  />
</View>
`;

exports[`ToDoListItem renders published assignment 1`] = `
<View>
  <Row
    accessible={true}
    disclosureIndicator={true}
    onPress={[Function]}
    renderImage={[Function]}
    testID="to-do.list.https://api.instructure.com/api/v1/path/to/html_url.row"
    title="Assignment 1"
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 14,
              "fontWeight": "600",
            },
            Object {
              "color": "#fff",
            },
          ]
        }
      >
        Course 1
      </Text>
      <Text
        style={
          Object {
            "color": "#8B969E",
            "fontSize": 13,
            "marginTop": 3,
          }
        }
      >
        Due May 31, 2037 at 11:59 PM
      </Text>
      <Text
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "white",
              "borderRadius": 9,
              "borderWidth": 1,
              "flex": 0,
              "fontSize": 11,
              "fontWeight": "600",
              "marginTop": 4,
              "overflow": "hidden",
              "paddingBottom": 1,
              "paddingLeft": 6,
              "paddingRight": 6,
              "paddingTop": 3,
            },
            Object {
              "borderColor": "#374A59",
              "color": "#374A59",
            },
          ]
        }
      >
        5 NEED GRADING
      </Text>
    </View>
  </Row>
  <AccessLine
    disableAppSpecificChecks={false}
    visible={true}
  />
</View>
`;

exports[`ToDoListItem renders quiz 1`] = `
<View>
  <Row
    accessible={true}
    disclosureIndicator={true}
    onPress={[Function]}
    renderImage={[Function]}
    testID="to-do.list.https://api.instructure.com/api/v1/path/to/html_url.row"
    title="Quiz 1"
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 14,
              "fontWeight": "600",
            },
            Object {
              "color": "#fff",
            },
          ]
        }
      >
        Course 1
      </Text>
      <Text
        style={
          Object {
            "color": "#8B969E",
            "fontSize": 13,
            "marginTop": 3,
          }
        }
      >
        Due Jan 23, 2013 at 11:59 PM
      </Text>
      <Text
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "white",
              "borderRadius": 9,
              "borderWidth": 1,
              "flex": 0,
              "fontSize": 11,
              "fontWeight": "600",
              "marginTop": 4,
              "overflow": "hidden",
              "paddingBottom": 1,
              "paddingLeft": 6,
              "paddingRight": 6,
              "paddingTop": 3,
            },
            Object {
              "borderColor": "#374A59",
              "color": "#374A59",
            },
          ]
        }
      >
        5 NEED GRADING
      </Text>
    </View>
  </Row>
  <AccessLine
    disableAppSpecificChecks={false}
    visible={true}
  />
</View>
`;

exports[`ToDoListItem renders unpublished assignment 1`] = `
<View>
  <Row
    accessible={true}
    disclosureIndicator={true}
    onPress={[Function]}
    renderImage={[Function]}
    testID="to-do.list.https://api.instructure.com/api/v1/path/to/html_url.row"
    title="Assignment 1"
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 14,
              "fontWeight": "600",
            },
            Object {
              "color": "#fff",
            },
          ]
        }
      >
        Course 1
      </Text>
      <Text
        style={
          Object {
            "color": "#8B969E",
            "fontSize": 13,
            "marginTop": 3,
          }
        }
      >
        Due May 31, 2037 at 11:59 PM
      </Text>
      <Text
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "white",
              "borderRadius": 9,
              "borderWidth": 1,
              "flex": 0,
              "fontSize": 11,
              "fontWeight": "600",
              "marginTop": 4,
              "overflow": "hidden",
              "paddingBottom": 1,
              "paddingLeft": 6,
              "paddingRight": 6,
              "paddingTop": 3,
            },
            Object {
              "borderColor": "#374A59",
              "color": "#374A59",
            },
          ]
        }
      >
        5 NEED GRADING
      </Text>
    </View>
  </Row>
  <AccessLine
    disableAppSpecificChecks={false}
    visible={false}
  />
</View>
`;
